resources/utilities/upload_vcf.py

```diff
@@ -0,0 +1,222 @@
+        'property#filename': filebase,
+        'property#reference_genome': reference_genome,
+        }
+
+    content_type = 'application/x-www-form-urlencoded'
+
+    with Request(method, href, content_type=content_type, body=urlencode(params)) as response:
+        assert response.status in [201, 202], \
+            'Unexpected response ({0}) from setting VCF properties'.format(response.status)
+
+        if response.status == 201:
+            href = response.getheader('Location')
+            logging.info('Created VCF object: {0}'.format(href))
+            return href
+        elif response.status == 202:
+            logging.info('Updated VCF object: {0}'.format(href))
+            return vcf_object
+
+def delete_old_vcf_file(record_id):
+    attachment_name = '{0}.vcf'.format(record_id)
+    href = '/rest/wikis/xwiki/spaces/data/pages/{0}/attachments/{1}'.format(record_id, attachment_name)
+    with Request('HEAD', href) as response:
+        if response.status in [200, 500]:
+            logging.info('Found linked old-style VCF: {0}'.format(attachment_name))
+        else:
+            return
+
+    with Request('DELETE', href) as response:
+        assert response.status in [204, 500], \
+            'Unexpected response ({0}) from deleting VCF file'.format(response.status)
+        logging.info('Deleted attachment: {0}'.format(attachment_name))
+
+    with Request('DELETE', href) as response:
+        assert response.status == 404, \
+            'Failed to delete existing old-style VCF file'.format(response.status)
+
+def upload_vcf_file(record_id, filename):
+    attachment_name = os.path.basename(filename)
+    href = '/rest/wikis/xwiki/spaces/data/pages/{0}/attachments/{1}'.format(record_id, attachment_name)
+    with open(filename) as file:
+        with Request('PUT', href, content_type='text/plain', body=file) as response:
+            assert response.status in [201, 202], \
+                'Unexpected response ({0}) from uploading VCF file'.format(response.status)
+            if response.status == 201:
+                logging.info('Created attachment: {0}'.format(attachment_name))
+            elif response.status == 202:
+                logging.info('Updated attachment: {0}'.format(attachment_name))
+
+def script(record_id, vcf_filename, credentials_file, server_name):
+    global SERVER
+    SERVER = server_name
+
+    load_credentials(credentials_file)
+    
+    consent_granted = get_consent(record_id)
+    if consent_granted == 'no':
+        logging.info('Record {0} has not granted genetic consent, can not upload VCF'.format(record_id))
+        return
+
+    vcf_object = get_vcf_object(record_id)
+
+    vcf_basename = os.path.basename(vcf_filename)
+    if vcf_object and vcf_object.get('headline', '') == vcf_basename:
+        # Filename already appears to be properly linked to record, skip
+        logging.info('Record {0} already has linked VCF file: {1}'.format(record_id, vcf_basename))
```

Comment:
How to re-upload a vcf file, which may be modified? I would like a `--force-reupload` option that will still upload the file, even if it has the same name.